### PR TITLE
Package hosted login deploy artifact

### PR DESCRIPTION
**Pull request by eng-cc**
## Summary
- Adds pull_request triggering for the hosted-login Linux build workflow so PRs can produce a deployable Linux x86_64 artifact.
- Uses checked-in viewer web dist for the deploy artifact to avoid hanging in an unnecessary web rebuild during this deploy slice.
- Broadens Rust build-input filters and narrows viewer triggers to checked-in dist inputs so the workflow does not package stale UI for source-only viewer changes.
- Creates a repo-owned PM task and records packaging/deployment evidence for both hosted-login test and production services.
- Adds the project-level environment matrix at `doc/engineering/governance/environment-lanes-and-inventory-2026-05-29.md`, covering test/prod lanes, hosted-login inventory, and testnet/mainnet claim boundaries.

## Validation
- `ruby -e 'require "yaml"; YAML.load_file(ARGV.fetch(0)); puts "workflow yaml ok"' .github/workflows/hosted-login-linux-build.yml`
- `ruby -e 'require "yaml"; YAML.load_file(ARGV.fetch(0)); puts "task yaml ok"' .pm/tasks/task_a0282b0e2c51476590c5de733f0d45ef.yaml`
- `./scripts/pm/lint.sh`
- `./scripts/doc-governance-check.sh`
- `git diff --check`
- GitHub Actions `Hosted Login Linux Build` run `26618151949`: success for packaged code artifact
- Artifact zip sha256: `c7b06708df969d449e6eb3935c9f4eb3bccc33505bb57ce1d6b5c1e14fa3d054`
- Artifact tar sha256: `e15f021166ec02a82148bfa0bf9437d049e83dc9e430a8b7bf3941d287d91115`

## Test Deployment Status
- Deployed on `39.104.205.67` at `/opt/oasis7/hosted-login-test/releases/20260529-1245-b3d684ab`.
- Active `current` points to `/opt/oasis7/hosted-login-test/releases/20260529-1245-b3d684ab`.
- `oasis7-hosted-login-test.service` is active; ports `4373`, `6511`, and `6523` are listening.
- External root smoke returned HTTP `200`.
- `POST /api/public/hosted-account/login/start` with the test mailbox returned `ok=true`, `challenge.delivery_mode=smtp`, and no `preview_code`.

## Production Deployment Status
- Deployed on `39.104.204.172` at `/opt/oasis7/hosted-login-prod/releases/20260529-1450-b3d684ab`.
- Active `current` points to `/opt/oasis7/hosted-login-prod/releases/20260529-1450-b3d684ab`.
- `oasis7-hosted-login-prod.service` is enabled and active; ports `4373`, `6511`, and `6523` are listening.
- External root smoke returned HTTP `200`.
- `POST /api/public/hosted-account/login/start` with the test mailbox returned `ok=true`, `challenge.delivery_mode=smtp`, and no `preview_code`.

## Environment Contract
- Every externally used flow should define a test lane and a prod lane before public claims.
- Hosted-login test/prod now have separate service names, roots, env files, and Tablestore tables.
- `public_testnet` and `mainnet` remain separate network-tier claims with their own gates; testnet service liveness is not mainnet readiness.

## Review Follow-up
- Addressed comments about Rust path filters, stale viewer UI packaging, non-existent root package paths, unused Node setup, and prod deployment acceptance scope.
- No secret values are included in this PR body, environment document, or PM execution log.

### CODE REVIEW

File: .github/workflows/hosted-login-linux-build.yml [hunk 1]
```diff
@@ -1,6 +1,16 @@
 name: Hosted Login Linux Build
 
 on:
+  pull_request:
+    branches: ["main"]
+    paths:
+      - ".github/workflows/hosted-login-linux-build.yml"
+      - "crates/oasis7/**"
+      - "crates/oasis7_viewer/**"
```
[the remark this answers is not in the document]
Comment (eng-cc) [reply .github/workflows/hosted-login-linux-build.yml]: Addressed in `6b58b5b66`: the PR path filters now include root `Cargo.toml` / `Cargo.lock`, `rust-toolchain.toml`, the `oasis7` path dependency crates used by this binary build, and the workspace vendor patch path.

File: .github/workflows/hosted-login-linux-build.yml [hunk 2]
```diff
@@ -87,9 +97,8 @@ jobs:
             --bin oasis7_game_launcher \
             --bin oasis7_viewer_live
 
-      - name: Build viewer web dist
+      - name: Stage viewer web dist
```
[the remark this answers is not in the document]
Comment (eng-cc) [reply .github/workflows/hosted-login-linux-build.yml:113]: Addressed in `6b58b5b66`: this workflow is now scoped to checked-in viewer dist inputs rather than broad `crates/oasis7_viewer/**` source changes, so a source-only viewer PR will not produce a stale deploy artifact from copied dist.

File: .github/workflows/hosted-login-linux-build.yml [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
+  pull_request:
+    branches: ["main"]
+    paths:
+      - ".github/workflows/hosted-login-linux-build.yml"
+      - "crates/oasis7/**"
+      - "crates/oasis7_viewer/**"
+      - "scripts/build-viewer-software-safe.sh"
+      - "scripts/copy-viewer-web-dist.sh"
+      - "package-lock.json"
+      - "package.json"
```
[the remark this answers is not in the document]
Comment (eng-cc) [reply .github/workflows/hosted-login-linux-build.yml]: Addressed in `6b58b5b66`: removed the non-existent root package paths and pointed at the real `crates/oasis7_viewer/package.json` and `crates/oasis7_viewer/package-lock.json` inputs.

File: .github/workflows/hosted-login-linux-build.yml [hunk 4]
```diff
@@ -87,9 +97,8 @@ jobs:
             --bin oasis7_game_launcher \
             --bin oasis7_viewer_live
 
-      - name: Build viewer web dist
+      - name: Stage viewer web dist
         run: |
-          ./scripts/build-viewer-software-safe.sh
           ./scripts/copy-viewer-web-dist.sh --dist-dir output/hosted-login/web
```
[the remark this answers is not in the document]
Comment (eng-cc) [reply .github/workflows/hosted-login-linux-build.yml:116]: Addressed in `6b58b5b66`: removed `actions/setup-node` and `npm ci` from this deploy artifact job since it stages checked-in viewer dist instead of rebuilding web assets.

File: .pm/tasks/task_a0282b0e2c51476590c5de733f0d45ef.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  - doc/p2p/blockchain/p2p-hosted-public-join-managed-identity-custody-2026-05-18.project.md
+doc_refs:
+  - doc/p2p/project.md
+related_prd:
+  - PRD-P2P-029-G
+acceptance:
+  - "Publish a PR for the smtp-only hosted login deployment slice."
+  - "Build a Linux x86_64 package from the current branch."
+  - "Deploy the package to 39.104.205.67 hosted-login-test with SMTP env and no preview_code in login/start."
+handoff_to: []
```
[the remark this answers is not in the document]
Comment (eng-cc) [reply .pm/tasks/task_a0282b0e2c51476590c5de733f0d45ef.yaml:22]: Addressed in `6b58b5b66`: updated the task title and acceptance list to include the 204 production deployment and the environment-lanes documentation scope.
